Problem: params not cached in object store

yihuang opened the pull request:
Solution:
- save the cost of repeated encoding/decoding during the block execution
- also migrate existing transient stores to object stores

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

yihuang commented on CHANGELOG.md:
```diff
@@ -50,6 +50,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [#447](https://github.com/crypto-org-chain/ethermint/pull/447) Deduct fee through virtual bank transfer.
 * (evm) [#448](https://github.com/crypto-org-chain/ethermint/pull/448) Refactor the evm transfer to be more efficient.
 * (evm) [#450](https://github.com/crypto-org-chain/ethermint/pull/450) Refactor transient stores to be compatible with parallel tx execution.
+* (evm) [#]() Migrate transient stores to object stores.
```
```suggestion
* (evm) [#454](https://github.com/crypto-org-chain/ethermint/pull/454) Migrate transient stores to object stores.
```

yihuang commented on x/feemarket/types/keys.go:
```diff
@@ -33,7 +36,17 @@ const (
 	KeyPrefixBlockGasWanted = []byte{prefixBlockGasWanted}
 )
+
+// KVStore key prefixes
```
```suggestion
// Object store key prefixes
```

mmsqe commented on x/evm/types/key.go:
```diff
@@ -30,9 +30,9 @@ const (
 	StoreKey = ModuleName
 
-	// TransientKey is the key to access the EVM transient store, that is reset
+	// ObjectKey is the key to access the EVM transient store, that is reset
```
it's object store right, should we change to `ObjectStoreKey` to be consistent?

yihuang replied:
done

Commit:
Problem: params not cached in object store

Solution:
- save the cost of repeated encoding/decoding during the block execution
- also migrate existing transient stores to object stores

Update CHANGELOG.md

Signed-off-by: yihuang <huang@crypto.com>

Update x/feemarket/types/keys.go

Signed-off-by: yihuang <huang@crypto.com>

renmae

fix test
